app/uk/gov/hmrc/eusubsidycompliancefrontend/actions/EscRequestVerifiedEmailActionBuilder.scala

```diff
@@ -58,18 +59,25 @@ class EscRequestCDSActionBuilder @Inject() (
     request: Request[A],
     block: AuthenticatedEscRequest[A] => Future[Result]
   ): Future[Result] =
-    authorised(Enrolment(CdsEnrolmentKey))
+    authorised()
       .retrieve[Option[Credentials] ~ Option[String] ~ Enrolments](
         Retrievals.credentials and Retrievals.groupIdentifier and Retrievals.allEnrolments
       ) {
         case Some(credentials) ~ Some(groupId) ~ enrolments =>
-          (enrolments.getEnrolment(EccEnrolmentKey), enrolments.getEnrolment(CdsEnrolmentKey)) match {
-            case (Some(eccEnrolment), Some(cdsEnrolment)) =>
-              val identifier = EscInitialRequestActionBuilder
-                .getIdentifier(eccEnrolment, cdsEnrolment, EnrolmentIdentifier)
-                .fold(throw new IllegalStateException("no eori provided"))(identity)
-              block(AuthenticatedEscRequest(credentials.providerId, groupId, request, EORI(identifier)))
-            case (None, Some(_)) => Redirect(appConfig.eccEscSubscribeUrl).toFuture
+          (enrolments.getEnrolment(EccEnrolmentKey)) match {
```

Comment:
I think this needs another pass to make it a little easier to read. Wondering if the for comprehension is buying us anything here - could just be replaced with a flatMap on the result of the result on the `getEmailVerification` call.